catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/ListApplication.java

```diff
@@ -0,0 +1,369 @@
+            List<String> ids = new LinkedList<>();
+            List<String> errorMessages = new LinkedList<>();
+
+            boolean isSplitSuccessful = false;
+
+            try (Stream<StorableResource> stream = createStream(createInfo, splitter)) {
+              isSplitSuccessful = true;
+              stream
+                  .sequential()
+                  .peek(storableResource -> appendMessageIfError(errorMessages, storableResource))
+                  .filter(storableResource -> !storableResource.isError())
+                  .forEach(
+                      storableResource ->
+                          storeAndClose(ids::add, errorMessages::add, storableResource));
+            } catch (IOException e) {
+              LOGGER.debug("Failed to split the incoming data. Trying the next splitter.", e);
+            }
+
+            if (isSplitSuccessful) {
+              /** TODO: DDF-3800 - Display these error messages in the UI. */
+              errorMessages.forEach(
+                  s -> LOGGER.debug(String.format("Unable to ingest split item: %s", s)));
+
+              response.header("Added-IDs", ids.stream().collect(Collectors.joining(",")));
+              break;
+            }
+          }
+
+          IOUtils.closeQuietly(createInfo.getStream());
+
+          return "";
+        });
+  }
+
+  private void storeAndClose(
+      Consumer<String> idConsumer,
+      Consumer<String> errorMessageConsumer,
+      StorableResource storableResource) {
+
+    try /*(StorableResource sr = storableResource)*/ {
+      store(getCreateInfo(storableResource), idConsumer, errorMessageConsumer);
+    } catch (IOException e) {
+      LOGGER.debug("Unable to create CreateInfo: ", e);
+    }
+  }
+
+  private Stream<StorableResource> createStream(CreateInfo createInfo, Splitter splitter)
+      throws IOException {
+    return splitter.split(createStorableResource(createInfo));
+  }
+
+  private StorableResource createStorableResource(CreateInfo createInfo) throws IOException {
+    return new StorableResource(createInfo.stream, createInfo.contentType, createInfo.filename);
+  }
+
+  private void appendMessageIfError(List<String> errorMessages, StorableResource storableResource) {
+    if (storableResource.isError()) {
+      errorMessages.add(storableResource.getErrorMessage());
+    }
+  }
+
+  private CreateInfo getCreateInfo(StorableResource storableResource) throws IOException {
+    CreateInfo createInfoItem = new CreateInfo();
+
+    createInfoItem.setFilename(storableResource.getFilename());
+    createInfoItem.setStream(storableResource.getInputStream());
+    createInfoItem.setContentType(
+        storableResource
+            .getMimeType()
+            .orElse(contentTypeFromFilename(storableResource.getFilename())));
+
+    return createInfoItem;
+  }
+
+  private CreateInfo parseAttachments(List<Part> contentParts) {
+
+    if (contentParts.size() == 1) {
```

Comment:
Consider adding a logging statement if the size is not 1

Reply:
done